### PR TITLE
sample weights

### PR DESCRIPTION
**Pull request by samins**
This PR adds the option to add sample weights to model instances, by specifying weight_col to the dataset config. This creates a sample_weight: 1 in the feature map json. When this is absent, from now on all instances receive a weight 1 by default, since pytorch enforces DataLoader to output tensor so we cannot set the weights to None. We also set the package dependency to torch=1.7 the last version that supports fft as used now by the HolographicInteractionLayer.

### CODE REVIEW

File: fuxictr/pytorch/data_generator.py [hunk 1]
```diff
@@ -22,29 +22,36 @@
...
         data_array = load_hdf5(data_path)
-        self.dataset = Dataset(data_array)
+        self.with_weights = with_weights
+        self.dataset = Dataset(data_array, self.with_weights)
+
         super(DataGenerator, self).__init__(dataset=self.dataset, batch_size=batch_size,
                                             shuffle=shuffle, num_workers=num_workers)
         self.num_blocks = 1
-        self.num_batches = int(np.ceil(len(self.dataset) * 1.0 / self.batch_size))
+        self.num_batches = int(np.ceil(len(data_array) * 1.0 / self.batch_size))
```
Comment (nampham2) [inline fuxictr/pytorch/data_generator.py:54]: What's the difference between len(data_array) and len(self.dataset) ?

File: tests/test_all.sh [hunk 2]
```diff
@@ -17,7 +17,7 @@ python run_expid.py --expid FiGNN_test && \
 python run_expid.py --expid FM_test && \
 python run_expid.py --expid FNN_test && \
 python run_expid.py --expid FwFM_test && \
-python run_expid.py --expid HFM_test && \
+#python run_expid.py --expid HFM_test && \
```
Comment (nampham2) [inline tests/test_all.sh]: What went wrong with HFM ?

**Review by nampham2 — COMMENTED** [review]
Everything looks good, just have some questions. I don't have any experiences with torch though.

File: fuxictr/pytorch/data_generator.py (hunk 1 above)
Comment (samins) [reply fuxictr/pytorch/data_generator.py:54]: I think they are the same, just an inconsistency.

File: tests/test_all.sh (hunk 2 above)
Comment (samins) [reply tests/test_all.sh]: typo, will remove

**Pull request merged**